lib/modules/dosomething/dosomething_campaign/includes/Campaign.php

```diff
@@ -0,0 +1,285 @@
+    $data['type'] = $image->type;
+    $data['title'] = $image->title;
+    $data['created_at'] = $image->created;
+    $data['updated_at'] = $image->changed;
+    $data['dark_background'] = (bool) dosomething_helpers_extract_field_data($image->field_dark_image);
+
+    // @TODO: consider reworking following function to return data instead of assigning it by reference to variable. Kind of confusing.
+    dosomething_campaign_load_image_url($data['sizes'], $image);
+    if ($data['sizes']['landscape']) {
+      $data['sizes']['landscape']['uri'] = dosomething_image_get_themed_image_url($image->nid, 'landscape', '1440x810');
+    }
+
+    if ($data['sizes']['square']) {
+      $data['sizes']['square']['uri'] = dosomething_image_get_themed_image_url($image->nid, 'square', '300x300');
+    }
+
+    return $data;
+  }
+
+
+  protected function getCoverImageAlt() {
+    $image_id = $this->variables['alt_image_campaign_cover_nid'];
+
+    if ($image_id) {
+      return $this->getCoverImage($image_id);
+    }
+    else {
+      return NULL;
+    }
+  }
+
+
+  protected function getFactData() {
+    $data = array();
+
+    $fact_fields = array('field_fact_problem', 'field_fact_solution');
+    $fact_vars = dosomething_fact_get_mutiple_fact_field_vars($this->node, $fact_fields);
+
+    if (!empty($fact_vars)) {
+      foreach ($fact_vars['facts'] as $index => $fact_data) {
+        $index = str_replace('field_', '', $index);
+
+        $data[$index]['fact'] = dosomething_helpers_issetor($fact_data['fact']);
+        $data[$index]['id'] = dosomething_helpers_issetor($fact_data['nid']);
+        $data[$index]['footnote'] = (int) dosomething_helpers_issetor($fact_data['footnotes']);
+        $data[$index]['source'] = dosomething_helpers_issetor($fact_data['sources']);
+
+      }
+
+      $data['sources'] = dosomething_helpers_issetor($fact_vars['sources']);
+
+      return $data;
+    }
+
+    return NULL;
+
+  }
+
+
+  protected function getIssue() {
+    // @TODO: Need to find out if this is allowed to be a field with multiple values...?
+    $issue_id = dosomething_helpers_extract_field_data($this->node->field_issue);
+
+    if ($issue_id) {
+      $issue = $this->getTaxonomyTerm($issue_id);
+
+      return $issue;
+    }
+
+    return NULL;
+  }
+
+
+  protected function getPrimaryCause() {
+    $cause_id = dosomething_helpers_extract_field_data($this->node->field_primary_cause);
+
+    if ($cause_id) {
+      $cause = $this->getTaxonomyTerm($cause_id);
+
+      return $cause;
+    }
+
+    return NULL;
+  }
+
+
+  protected function getTags() {
+    $data = array();
+
+    $tag_ids = dosomething_helpers_extract_field_data($this->node->field_tags);
+
+    if ($tag_ids) {
+      foreach ($tag_ids as $tid) {
+        $data[] = $this->getTaxonomyTerm($tid);
+      }
+
+      return $data;
+    }
+
+    return NULL;
+  }
+
+
+  protected function getTaxonomyTerm($tid) {
+    $data = array();
+
+    $taxonomy = taxonomy_term_load($tid);
```

Comment:
This is something to consider. The loaded `$node` object will already have the taxonomy term object loaded. What I'm not sure of is whether or not this call will perform a secondary `taxonomy_term_load` for each term.  I suppose we can leave it for now and if performance is an issue, this would def be something to look into...